src/OpenTelemetry/Logs/LogRecord.cs

```diff
@@ -47,71 +50,102 @@ public sealed class LogRecord
             Exception? exception,
             IReadOnlyList<KeyValuePair<string, object?>>? stateValues)
         {
-            this.ScopeProvider = scopeProvider;
-
-            var activity = Activity.Current;
-            if (activity != null)
+            this.Data = new(Activity.Current)
             {
-                this.TraceId = activity.TraceId;
-                this.SpanId = activity.SpanId;
-                this.TraceState = activity.TraceStateString;
-                this.TraceFlags = activity.ActivityTraceFlags;
-            }
+                TimestampBacking = timestamp,
```

Comment:
Not using the public Timestamp to avoid small perf hit to check Kind?

Reply:
Ya 🤣